app/services/articles/feeds/weighted_query_strategy.rb

```diff
@@ -205,6 +209,45 @@ class WeightedQueryStrategy
                 AND followed_tags.follower_id = :user_id
                 AND followed_tags.explicit_points >= 0"]
         },
+        # Weight to give for the number of intersecting tags the given
+        # user follows with a positive score and the article has.
+        matching_positive_tag_intersection_count_factor: {
+          clause: "COUNT(followed_tags.follower_id)",
+          cases: [[0, 0.86], [1, 0.94], [2, 0.98]],
+          fallback: 1,
+          requires_user: true,
+          joins: ["LEFT OUTER JOIN taggings
+            ON taggings.taggable_id = articles.id
+              AND taggable_type = 'Article'",
+                  "INNER JOIN tags
+              ON taggings.tag_id = tags.id",
+                  "LEFT OUTER JOIN follows AS followed_tags
+              ON tags.id = followed_tags.followable_id
+                AND followed_tags.followable_type = 'ActsAsTaggableOn::Tag'
+                AND followed_tags.follower_type = 'User'
+                AND followed_tags.follower_id = :user_id
+                AND followed_tags.explicit_points >= 0"]
+        },
+        # Weight to give for the number of intersecting tags the given user follows with a negative
+        # score and the article has.
+        matching_negative_tag_intersection_count_factor: {
+          # NOTE: We're not looking at the actual negative points but the number of negative tags.
+          clause: "COUNT(negative_followed_tags.id)",
+          cases: [[0, 1], [1, 0.9], [2, 0.75], [3, 0.55], [4, 0.3]],
```

Comment:
This is the reverse of the last lever, discounting articles with many tags I purposefully anti-followed.

Same consideration (users that curate an anti-follow list benefit most, users with no negative tag follow weightings will be unaffected).